--- /dev/null
+++ b/src/main/java/me/earthme/luminol/config/modules/fixes/FoliaPOIAccessOffRegionFixConfig.java
@@ -1,0 +_,25 @@
+package me.earthme.luminol.config.modules.fixes;
+
+import me.earthme.luminol.config.ConfigInfo;
+import me.earthme.luminol.config.EnumConfigCategory;
+import me.earthme.luminol.config.IConfigModule;
+
+public class FoliaPOIAccessOffRegionFixConfig implements IConfigModule {
+    @ConfigInfo(baseName = "enabled", comments =
+            """
+                    The POIManager of folia has something which has not been patched\s
+                    for regionized ticking and these would trigger the async catcher\s
+                    and make the server crash.If you would like to prevent it and didn't\s
+                    mind the side effect(currently unknown), you can enable this""")
+    public static boolean enabled = false;
+
+    @Override
+    public EnumConfigCategory getCategory() {
+        return EnumConfigCategory.FIXES;
+    }
+
+    @Override
+    public String getBaseName() {
+        return "folia.fix_poi_access_off_region";
+    }
+}
