--- /dev/null
+++ b/src/main/java/me/earthme/luminol/config/modules/fixes/FoliaEntityMovingFixConfig.java
@@ -1,0 +_,28 @@
+package me.earthme.luminol.config.modules.fixes;
+
+import me.earthme.luminol.config.ConfigInfo;
+import me.earthme.luminol.config.EnumConfigCategory;
+import me.earthme.luminol.config.IConfigModule;
+
+public class FoliaEntityMovingFixConfig implements IConfigModule {
+    @ConfigInfo(baseName = "enabled", comments =
+            """
+                    A simple fix of a issue on folia\s
+                    (Some times the entity would\s
+                    have a large moment that cross the\s
+                    different tick regions and it would\s
+                    make the server crashed)  but sometimes it might doesn't work""")
+    public static boolean enabled = false;
+    @ConfigInfo(baseName = "warn_on_detected")
+    public static boolean warnOnDetected = true;
+
+    @Override
+    public EnumConfigCategory getCategory() {
+        return EnumConfigCategory.FIXES;
+    }
+
+    @Override
+    public String getBaseName() {
+        return "folia.fix_high_velocity_issue";
+    }
+}
