--- /dev/null
+++ b/src/main/java/me/earthme/luminol/commands/MembarCommand.java
@@ -1,0 +_,47 @@
+package me.earthme.luminol.commands;
+
+import me.earthme.luminol.config.modules.misc.MembarConfig;
+import me.earthme.luminol.functions.GlobalServerMemoryBar;
+import net.kyori.adventure.text.Component;
+import net.kyori.adventure.text.format.TextColor;
+import org.bukkit.command.Command;
+import org.bukkit.command.CommandSender;
+import org.bukkit.entity.Player;
+import org.jetbrains.annotations.NotNull;
+
+public class MembarCommand extends Command {
+    public MembarCommand(@NotNull String name) {
+        super(name);
+        this.setPermission("luminol.commands.membar");
+        this.setDescription("Show the memory usage through a bossbar");
+        this.setUsage("/membar");
+    }
+
+    @Override
+    public boolean execute(@NotNull CommandSender sender, @NotNull String commandLabel, @NotNull String[] args) {
+        if (!testPermission(sender)){
+            return true;
+        }
+
+        if (!MembarConfig.memoryBarEnabled){
+            sender.sendMessage(Component.text("Membar was already disabled!").color(TextColor.color(255,0,0)));
+            return true;
+        }
+
+        if (!(sender instanceof Player player)){
+            sender.sendMessage(Component.text("Only player can use this command!").color(TextColor.color(255,0,0)));
+            return true;
+        }
+
+        if (GlobalServerMemoryBar.isPlayerVisible(player)) {
+            player.sendMessage(Component.text("Disabled mem bar").color(TextColor.color(0,255,0)));
+            GlobalServerMemoryBar.setVisibilityForPlayer(player,false);
+            return true;
+        }
+
+        player.sendMessage(Component.text("Enabled mem bar").color(TextColor.color(0,255,0)));
+        GlobalServerMemoryBar.setVisibilityForPlayer(player,true);
+
+        return true;
+    }
+}
