--- /dev/null
+++ b/src/main/java/me/earthme/luminol/config/modules/misc/TpsBarConfig.java
@@ -1,0 +_,51 @@
+package me.earthme.luminol.config.modules.misc;
+
+import com.electronwill.nightconfig.core.file.CommentedFileConfig;
+import me.earthme.luminol.commands.TpsBarCommand;
+import me.earthme.luminol.config.*;
+import me.earthme.luminol.functions.GlobalServerTpsBar;
+import org.bukkit.Bukkit;
+
+import java.util.List;
+
+public class TpsBarConfig implements IConfigModule {
+    @ConfigInfo(baseName = "enabled")
+    public static boolean tpsbarEnabled = false;
+    @ConfigInfo(baseName = "format")
+    public static String tpsBarFormat = "<gray>TPS<yellow>:</yellow> <tps> MSPT<yellow>:</yellow> <mspt> Ping<yellow>:</yellow> <ping>ms ChunkHot<yellow>:</yellow> <chunkhot>";
+    @ConfigInfo(baseName = "tps_color_list")
+    public static List<String> tpsColors = List.of("GREEN","YELLOW","RED","PURPLE");
+    @ConfigInfo(baseName = "ping_color_list")
+    public static List<String> pingColors = List.of("GREEN","YELLOW","RED","PURPLE");
+    @ConfigInfo(baseName = "chunkhot_color_list")
+    public static List<String> chunkHotColors = List.of("GREEN","YELLOW","RED","PURPLE");
+    @ConfigInfo(baseName = "update_interval_ticks")
+    public static int updateInterval = 15;
+
+    @DoNotLoad
+    private static boolean inited = false;
+
+    @Override
+    public EnumConfigCategory getCategory() {
+        return EnumConfigCategory.MISC;
+    }
+
+    @Override
+    public String getBaseName() {
+        return "tpsbar";
+    }
+
+    @Override
+    public void onLoaded(CommentedFileConfig configInstance){
+        if (tpsbarEnabled){
+            GlobalServerTpsBar.init();
+        }else{
+            GlobalServerTpsBar.cancelBarUpdateTask();
+        }
+
+        if (!inited){
+            Bukkit.getCommandMap().register("tpsbar","luminol",new TpsBarCommand("tpsbar"));
+            inited = true;
+        }
+    }
+}
