From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Quentin GAVOILLE <quentin.gavoille@gmail.com>
Date: Tue, 28 Jan 2025 22:49:49 +0100
Subject: [PATCH] Re-add enderpearl chunkloading


diff --git a/net/minecraft/world/entity/projectile/ThrownEnderpearl.java b/net/minecraft/world/entity/projectile/ThrownEnderpearl.java
index f66d16d6efb1322481f6fed805a008bfc6e5335d..c0b4ca6c46923d86c428a4d65a7f6c854edcf7dc 100644
--- a/net/minecraft/world/entity/projectile/ThrownEnderpearl.java
+++ b/net/minecraft/world/entity/projectile/ThrownEnderpearl.java
@@ -284,6 +284,7 @@ public class ThrownEnderpearl extends ThrowableItemProjectile {
             super.tick();
         }
 
+        Entity entity = this.getOwnerRaw();
         if (this.isAlive()) {
             BlockPos blockPos = BlockPos.containing(this.position());
             if ((
@@ -291,7 +292,7 @@ public class ThrownEnderpearl extends ThrowableItemProjectile {
                         || sectionPosX != SectionPos.blockToSectionCoord(blockPos.getX())
                         || sectionPosZ != SectionPos.blockToSectionCoord(blockPos.getZ())
                 )
-                && owner instanceof ServerPlayer serverPlayer1) {
+                && entity instanceof ServerPlayer serverPlayer1) {
                 this.ticketTimer = serverPlayer1.registerAndUpdateEnderPearlTicket(this);
             }
         }
